### PR TITLE
Added getter for glprogram::programID

### PR DESCRIPTION
**Pull request by Quaker762**
Added a way to get the programID from a shader program instance.

### CODE REVIEW

**Comment by Quaker762** [comment]
This should be merged before #114

File: include/SH3/system/glprogram.hpp [hunk 1]
```diff
@@ -73,6 +73,14 @@ namespace sh3_gl
          */
         void Unbind();
 
+        /**
+         *  Gets the OpenGL generated Program ID from instance of this (shader) class so that
+         *  we can send/get data from outside of it.
+         *
+         *  return @ref programID The ID of this shader program.
+         */
+        inline GLuint GetProgramID(){return programID;}
```
Comment (z33ky) [inline include/SH3/system/glprogram.hpp]: `inline` gets implied, so we don't need to specify that.
This function can be `const` though, which cannot be inferred by the compiler.

File: include/SH3/system/glprogram.hpp [hunk 2]
```diff
@@ -73,6 +73,14 @@ namespace sh3_gl
          */
         void Unbind();
 
+        /**
+         *  Gets the OpenGL generated Program ID from instance of this (shader) class so that
```
Comment (z33ky) [inline include/SH3/system/glprogram.hpp]: The first sentence is a short description. I'd write something like "Gets the Program ID.". Then have an empty line, followed by the long description, if any is necessary.

**Review by z33ky — CHANGES_REQUESTED** [review]

File: include/SH3/system/glprogram.hpp (hunk 1 above)
Comment (Quaker762) [reply include/SH3/system/glprogram.hpp]: Oh I see, so the compiler is smart enough to know that it should be inlined?? I've been reading a bit of old C++ lately so excuse the verbose syntax.

Comment (z33ky) [reply include/SH3/system/glprogram.hpp]: Methods that are defined in the class definition are treated as specified with inline.

But the inline specifier is mostly just required to specify appropriate linkage. Compilers nowadays use heuristics based on the body and caller locations to determine when to inline functions. Having the definition available when it's called is essential for inlining though, unless you're compiling with LTO.

**Pull request merged**